Add documentation for custom clustering and multiplex_filtering into webpages (#923)
xia2/xia2: doc/sphinx/multiplex/intensity_based_clustering.rst

++++++++++++++++++++++++++
Intensity-based Clustering
++++++++++++++++++++++++++
---------------------------------------
Scaling and merging of dataset clusters
---------------------------------------
*To trigger further scaling and merging of clusters, set the xia2.multiplex option*
``output_clusters=True``.
Note that the clustering options described here can also be run standalone with the ``xia2.cluster_analysis`` program, using the xia2.multiplex scaled
data files (``scaled.expt``, ``scaled.refl``) as input, but each cluster will not be further scaled and merged.

Dataset clustering can be performed based on a hierarchical dendrogram analysis (``clustering.method=hierarchical``)
or on density-based analysis of the cosym coordinates (``clustering.method=coordinate``). See the publication at https://doi.org/10.1107/S2059798325004589 for
more details of the density-based analysis.

> Both methods share some common criteria when deciding which clusters to be output. These are tailorable via commandline input.

Reviewer: ```suggestion
Both methods share some common criteria when deciding which clusters to be output. These are tailorable via command-line input.
```

.. image:: ../figures/multiplex-dendrogram.png

As shown in this image, the height of a cluster is the point at which it separates into two smaller clusters in the dendrogram. So in this
example, the largest cluster splits at a height of 0.072, and these two clusters each split again around a height of 0.018.
So if clustering was run with ``max_output_clusters=2``, one would get the two largest clusters (clusters 13 and 14 here). But if the ``max_cluster_height`` were set to a value just below 0.018, one would obtain
the two smaller clusters with heights 0.0075 and 0.0032 (clusters 12 and 11).

After the clusters have been selected, each cluster is then individually scaled and merged.

-------------------------
Distinct cluster analysis
-------------------------
The analysis described above does not guarantee that clusters will not share some individual datasets; this is not necessarily a problem, a typical use case for xia2.multiplex is generating a number
of similar clusters containing similar datasets, to compare the effects of removing a number of outlier datasets on the overall statistics.

> An alternative use case is where one is interested in distinct clusters of datasets, i.e. clusters that do not share any individual datasets, which may correspond to distinct crystal strutures

Reviewer: ```suggestion
An alternative use case is where one is interested in distinct clusters of datasets, i.e. clusters that do not share any individual datasets, which may correspond to distinct crystal structures
```

(this is the kind of clustering structure demonstrated in the image above). While this can be handled using the options above for small or simple cases, for large datasets the dendrogram structure

> becomes very complex, with many dendrogram subtructures and branches. As such, the options above do not guarantee selection and evaluation of distinct clusters in a timely manner.

Reviewer: ```suggestion
becomes very complex, with many dendrogram substructures and branches. As such, the options above do not guarantee selection and evaluation of distinct clusters in a timely manner.
```
